shorebird_code_push/lib/src/updater.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:ffi' as ffi;
+
+import 'package:ffi/ffi.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+import 'package:platform/platform.dart';
+import 'package:shorebird_code_push/src/generated/updater_bindings.g.dart';
+
+/// {@template updater}
+/// A wrapper around the generated [UpdaterBindings] that translates ffi types
+/// into easier to use Dart types.
+/// {@endtemplate}
+class Updater {
+  /// Creates an [Updater] instance using the currently loaded dynamic library.
+  Updater.init() {
+    bindings = UpdaterBindings(ffi.DynamicLibrary.process());
+  }
+
+  /// Creates an [Updater] instance using the dynamic library at the given
+  /// [directory].
+  ///
+  /// The name of the library is determined by the current OS:
+  ///   macOS:   lib[name].dylib
+  ///   Windows: [name].dll
+  ///   Linux:   lib[name].so
+  Updater.initWithLibrary({
```

Comment:
This is only needed in a pure Dart environment (which we doubt we support anymore unless you've resurrected it).

Reply:
I'll remove for now. It'll always be in commit history if we want to resurrect.